stack.go

```diff
@@ -130,34 +128,58 @@ func (c Call) Format(s fmt.State, verb rune) {
 // PC returns the program counter for this call frame; multiple frames may
 // have the same PC value.
 func (c Call) PC() uintptr {
-	return c.pc
+	frame := c.frame()
+	return frame.PC
 }
 
 // name returns the import path qualified name of the function containing the
 // call.
 func (c Call) name() string {
-	if c.fn == nil {
+	frame := c.frame()
+
+	if frame.Func == nil {
 		return "???"
 	}
-	return c.fn.Name()
+
+	return frame.Function
 }
 
 func (c Call) file() string {
-	if c.fn == nil {
+	frame := c.frame()
+
+	if frame.Func == nil {
 		return "???"
 	}
-	file, _ := c.fn.FileLine(c.pc)
+
+	file := frame.File
 	return file
 }
 
 func (c Call) line() int {
-	if c.fn == nil {
+	frame := c.frame()
+
+	if frame.Func == nil {
 		return 0
 	}
-	_, line := c.fn.FileLine(c.pc)
+
+	line := frame.Line
 	return line
 }
 
+// name returns the import path qualified name of the function containing the
+// call.
+func (c Call) frame() runtime.Frame {
+	frames := runtime.CallersFrames(c.pcs[:])
+	frame, _ := frames.Next()
+
+	if frame.Function == "runtime.sigpanic" {
+		frames = runtime.CallersFrames([]uintptr{c.pcs[1] + 1})
+	}
+
+	frame, _ = frames.Next()
+	return frame
+}
```

Comment:
`Call.frame` is doing too much. `frames.Next` already handles the sigpanic checks for us, so I believe `frame` should look more like this:

```go
func (c Call) frame() runtime.Frame {
	frames := runtime.CallersFrames(c.pcs[:])
	frame, more := frames.Next()
	if !more {
		return frame
	}
	frame, _ = frames.Next()
	return frame
}
```

Also, the doc comment doesn't match the method.